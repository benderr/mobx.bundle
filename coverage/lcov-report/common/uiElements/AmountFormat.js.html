<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for common/uiElements/AmountFormat.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">common/uiElements</a> AmountFormat.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/16</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import PropTypes from 'prop-types'
import {isEmpty} from '../validators'
import accounting from 'accounting';
&nbsp;
const CurrencySymbol = <span class="fstat-no" title="function not covered" >({</span>value}) =&gt; {
<span class="cstat-no" title="statement not covered" >	if (value == 'RUR')</span>
<span class="cstat-no" title="statement not covered" >		return (&lt;span class="cur ruble"&gt;&lt;span&gt;р.&lt;/span&gt;&lt;/span&gt;);</span>
<span class="cstat-no" title="statement not covered" >	if (value == 'USD')</span>
<span class="cstat-no" title="statement not covered" >		return (&lt;span class="cur dollar"&gt;&lt;span&gt;$&lt;/span&gt;&lt;/span&gt;);</span>
<span class="cstat-no" title="statement not covered" >	if (value == 'EUR')</span>
<span class="cstat-no" title="statement not covered" >		return (&lt;span class="cur euro"&gt;&lt;span&gt;€&lt;/span&gt;&lt;/span&gt;);</span>
<span class="cstat-no" title="statement not covered" >	return null;</span>
};
&nbsp;
const AmountFormat = <span class="fstat-no" title="function not covered" >({</span>value, currency = <span class="branch-0 cbranch-no" title="branch not covered" >'RUR',</span> def = <span class="branch-0 cbranch-no" title="branch not covered" >''}</span>) =&gt; {
<span class="cstat-no" title="statement not covered" >	if (isEmpty(value))</span>
<span class="cstat-no" title="statement not covered" >		return def ? (&lt;span&gt;{def}&lt;/span&gt;) : null;</span>
	const val = <span class="cstat-no" title="statement not covered" >parseFloat(cleanValue(value));</span>
	const formatted = <span class="cstat-no" title="statement not covered" >!isNaN(val) ? accounting.formatNumber(val, 2, " ") : def;</span>
<span class="cstat-no" title="statement not covered" >	return (&lt;span&gt;{formatted}&amp;nbsp;&lt;CurrencySymbol value={currency}/&gt;&lt;/span&gt;);</span>
};
&nbsp;
function <span class="fstat-no" title="function not covered" >cleanValue(</span>val) {
<span class="cstat-no" title="statement not covered" >	return val.replace ? val.replace(/[^0-9\.,]+/g, '').replace(',', '.') : val;</span>
}
&nbsp;
AmountFormat.propTypes = {
	value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
	currency: PropTypes.oneOf(['RUR', 'USD', 'EUR']),
	def: PropTypes.string
};
&nbsp;
export  default AmountFormat;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 17:38:13 GMT+0500 (+05)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
