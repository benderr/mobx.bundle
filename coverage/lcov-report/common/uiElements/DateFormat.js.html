<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for common/uiElements/DateFormat.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">common/uiElements</a> DateFormat.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/10</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import dateHelper from '../helpers/dateHelper'
import PropTypes from 'prop-types'
const DateFormat = <span class="fstat-no" title="function not covered" >({</span>value, format = <span class="branch-0 cbranch-no" title="branch not covered" >"clever",</span> def = <span class="branch-0 cbranch-no" title="branch not covered" >''}</span>) =&gt; {
	let formatDate = <span class="cstat-no" title="statement not covered" >value;</span>
<span class="cstat-no" title="statement not covered" >	if (value &amp;&amp; typeof value === 'string') {</span>
<span class="cstat-no" title="statement not covered" >		formatDate = dateHelper.stringToDate(value);</span>
<span class="cstat-no" title="statement not covered" >		if (isNaN(formatDate.getTime()))</span>
<span class="cstat-no" title="statement not covered" >			formatDate = new Date(value);</span>
	}
&nbsp;
<span class="cstat-no" title="statement not covered" >	if (formatDate &amp;&amp; formatDate instanceof Date)</span>
<span class="cstat-no" title="statement not covered" >		return (&lt;span&gt;{dateHelper.dateFormat(formatDate, format)}&lt;/span&gt;);</span>
<span class="cstat-no" title="statement not covered" >	return (&lt;span&gt;{def}&lt;/span&gt;);</span>
};
&nbsp;
DateFormat.propTypes = {
	value: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(Date)]),
	format: PropTypes.string,
	def: PropTypes.string
};
&nbsp;
export  default DateFormat;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 17:38:13 GMT+0500 (+05)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
