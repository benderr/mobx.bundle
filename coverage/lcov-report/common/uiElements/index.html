<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for common/uiElements</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> common/uiElements
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.15% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>43/354</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">2.11% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/285</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">1.9% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/105</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.03% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>42/349</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="AmountFormat.js"><a href="AmountFormat.js.html">AmountFormat.js</a></td>
	<td data-value="18.75" class="pic low"><div class="chart"><div class="cover-fill" style="width: 18%;"></div><div class="cover-empty" style="width:82%;"></div></div></td>
	<td data-value="18.75" class="pct low">18.75%</td>
	<td data-value="16" class="abs low">3/16</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="16" class="abs low">0/16</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="18.75" class="pct low">18.75%</td>
	<td data-value="16" class="abs low">3/16</td>
	</tr>

<tr>
	<td class="file low" data-value="AmountInput.js"><a href="AmountInput.js.html">AmountInput.js</a></td>
	<td data-value="2.33" class="pic low"><div class="chart"><div class="cover-fill" style="width: 2%;"></div><div class="cover-empty" style="width:98%;"></div></div></td>
	<td data-value="2.33" class="pct low">2.33%</td>
	<td data-value="86" class="abs low">2/86</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="75" class="abs low">0/75</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="16" class="abs low">0/16</td>
	<td data-value="2.33" class="pct low">2.33%</td>
	<td data-value="86" class="abs low">2/86</td>
	</tr>

<tr>
	<td class="file medium" data-value="Button.js"><a href="Button.js.html">Button.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	</tr>

<tr>
	<td class="file low" data-value="DateFormat.js"><a href="DateFormat.js.html">DateFormat.js</a></td>
	<td data-value="20" class="pic low"><div class="chart"><div class="cover-fill" style="width: 20%;"></div><div class="cover-empty" style="width:80%;"></div></div></td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="10" class="abs low">2/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="10" class="abs low">2/10</td>
	</tr>

<tr>
	<td class="file low" data-value="DatePicker.js"><a href="DatePicker.js.html">DatePicker.js</a></td>
	<td data-value="5.56" class="pic low"><div class="chart"><div class="cover-fill" style="width: 5%;"></div><div class="cover-empty" style="width:95%;"></div></div></td>
	<td data-value="5.56" class="pct low">5.56%</td>
	<td data-value="36" class="abs low">2/36</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="31" class="abs low">0/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="5.56" class="pct low">5.56%</td>
	<td data-value="36" class="abs low">2/36</td>
	</tr>

<tr>
	<td class="file low" data-value="DatePickerRange.js"><a href="DatePickerRange.js.html">DatePickerRange.js</a></td>
	<td data-value="4.26" class="pic low"><div class="chart"><div class="cover-fill" style="width: 4%;"></div><div class="cover-empty" style="width:96%;"></div></div></td>
	<td data-value="4.26" class="pct low">4.26%</td>
	<td data-value="47" class="abs low">2/47</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="22" class="abs low">0/22</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="17" class="abs low">0/17</td>
	<td data-value="4.26" class="pct low">4.26%</td>
	<td data-value="47" class="abs low">2/47</td>
	</tr>

<tr>
	<td class="file low" data-value="Drop.jsx"><a href="Drop.jsx.html">Drop.jsx</a></td>
	<td data-value="4.17" class="pic low"><div class="chart"><div class="cover-fill" style="width: 4%;"></div><div class="cover-empty" style="width:96%;"></div></div></td>
	<td data-value="4.17" class="pct low">4.17%</td>
	<td data-value="48" class="abs low">2/48</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="16" class="abs low">0/16</td>
	<td data-value="4.17" class="pct low">4.17%</td>
	<td data-value="48" class="abs low">2/48</td>
	</tr>

<tr>
	<td class="file high" data-value="InfinateScroll.jsx"><a href="InfinateScroll.jsx.html">InfinateScroll.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="LoaderBlock.js"><a href="LoaderBlock.js.html">LoaderBlock.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file medium" data-value="LoaderPanel.js"><a href="LoaderPanel.js.html">LoaderPanel.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	</tr>

<tr>
	<td class="file low" data-value="NumberFormat.js"><a href="NumberFormat.js.html">NumberFormat.js</a></td>
	<td data-value="20" class="pic low"><div class="chart"><div class="cover-fill" style="width: 20%;"></div><div class="cover-empty" style="width:80%;"></div></div></td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="15" class="abs low">3/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="15" class="abs low">3/15</td>
	</tr>

<tr>
	<td class="file low" data-value="NumberInput.js"><a href="NumberInput.js.html">NumberInput.js</a></td>
	<td data-value="3.85" class="pic low"><div class="chart"><div class="cover-fill" style="width: 3%;"></div><div class="cover-empty" style="width:97%;"></div></div></td>
	<td data-value="3.85" class="pct low">3.85%</td>
	<td data-value="26" class="abs low">1/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="20" class="abs low">0/20</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="3.85" class="pct low">3.85%</td>
	<td data-value="26" class="abs low">1/26</td>
	</tr>

<tr>
	<td class="file medium" data-value="PrimaryButton.js"><a href="PrimaryButton.js.html">PrimaryButton.js</a></td>
	<td data-value="66.67" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 66%;"></div><div class="cover-empty" style="width:34%;"></div></div></td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	</tr>

<tr>
	<td class="file low" data-value="Recaptcha.js"><a href="Recaptcha.js.html">Recaptcha.js</a></td>
	<td data-value="16.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 16%;"></div><div class="cover-empty" style="width:84%;"></div></div></td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="30" class="abs low">5/30</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="17.24" class="pct low">17.24%</td>
	<td data-value="29" class="abs low">5/29</td>
	</tr>

<tr>
	<td class="file low" data-value="Select.js"><a href="Select.js.html">Select.js</a></td>
	<td data-value="10" class="pic low"><div class="chart"><div class="cover-fill" style="width: 10%;"></div><div class="cover-empty" style="width:90%;"></div></div></td>
	<td data-value="10" class="pct low">10%</td>
	<td data-value="10" class="abs low">1/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="8" class="abs low">1/8</td>
	</tr>

<tr>
	<td class="file low" data-value="SortLink.js"><a href="SortLink.js.html">SortLink.js</a></td>
	<td data-value="33.33" class="pic low"><div class="chart"><div class="cover-fill" style="width: 33%;"></div><div class="cover-empty" style="width:67%;"></div></div></td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	</tr>

<tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 17:38:13 GMT+0500 (+05)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
