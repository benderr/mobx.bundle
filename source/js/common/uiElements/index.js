import PrimaryButton from './PrimaryButton';
import Button from './Button';
import LoaderBlock from './LoaderBlock';
import Select from './Select';
import AmountInput from './AmountInput';
import Recaptcha from './Recaptcha';
import InfinateScroll from './InfinateScroll';
import ConfirmPopupService from './Dialogs/ConfirmPopupService';
import * as Notify from './Notify';
import Drop from './Drop';
import DateFormat from './DateFormat';

export {
	PrimaryButton,
	LoaderBlock,
	Select,
	AmountInput,
	Recaptcha,
	InfinateScroll,
	ConfirmPopupService,
	Drop,
	Button,
	Notify,
	DateFormat
}