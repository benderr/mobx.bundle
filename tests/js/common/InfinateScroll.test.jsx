/**
 * Created by RobertSabiryanov on 12.07.17.
 */
import React from 'react';
import {InfinateScroll} from 'common/uiElements';
import renderer from 'react-test-renderer';

const itemsLength = 51;
let loadNext =()=>{

};

test('Infinate default class on loading=false', () => {

	const loading = false;

	const component = renderer.create(
		<InfinateScroll loadNext={loadNext} totalCount={itemsLength} listLength={50} loading={loading}/>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Infinate loading class on loading=true', () => {
    const loading = true;
    const component = renderer.create(
		<InfinateScroll loadNext={loadNext} totalCount={itemsLength} listLength={50} loading={loading}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

//todo про jest -u