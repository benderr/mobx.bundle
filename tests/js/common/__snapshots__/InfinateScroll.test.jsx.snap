// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Infinate default class on loading=false 1`] = `
<div
  className="table_row"
>
  <span
    style={
      Object {
        "fontSize": 0,
      }
    }
  />
</div>
`;

exports[`Infinate loading class on loading=true 1`] = `
<div
  className="table_row loading_block  loading_block__h40"
>
  <span
    style={
      Object {
        "fontSize": 0,
      }
    }
  />
</div>
`;
